src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpectatorView.cs

```diff
@@ -146,6 +163,14 @@ private void Awake()
             SetupRecordingService();
         }
 
+        private void OnDestroy()
+        {
+            if (Role == Role.Spectator)
+            {
+                SpatialCoordinateSystemManager.Instance.ParticipantConnected -= OnParticipantConnected;

```

Comment:
>SpatialCoordinateSystemManager.Instance.ParticipantConnected -= OnParticipantConnected; [](start = 16, length = 87)

nit: Could do this outside of the if statement, as it will be a no-op if not subscribed